packages/react-devtools-shared/src/hook.js

```diff
@@ -21,6 +21,25 @@ export function installHook(target: any): DevToolsHook | null {
     return null;
   }
 
+  function getMainWindow(targetWindow: any): any {
+    if (!canAccessParentWindow(targetWindow) || isMainWindow(targetWindow)) {
+      return targetWindow;
+    }
+    return getMainWindow(targetWindow.parent);
+  }
+
+  function isMainWindow(targetWindow: any): boolean {
+    return targetWindow.self === targetWindow.top;
```

Comment:
Why `window.self` rather than just `window`?

Reply:
It is the same but `self` can work with worker(it is more flexible)

Reply:
Ok just wanted to confirm I wasn't missing anything. 😄